controllers/hostedcontrolplane/synthetics.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return fmt.Errorf("error calling determineDynatraceClusterRegionId: %v", err)
+	}
+
+	locationId, err := dynatraceApiClient.GetLocationEntityIdFromDynatrace(dynatraceClusterRegionName, monitorLocationType)
+	if err != nil {
+		return fmt.Errorf("error calling GetLocationEntityIdFromDynatrace: %v", err)
+	}
+
+	clusterID := hostedcontrolplane.Spec.ClusterID
+	if clusterID == "" {
+		return fmt.Errorf("hostedcontrolplane has empty .Spec.ClusterID field")
+	}
+
+	// Check for existing monitors
+	monitors, err := dynatraceApiClient.ListDynatraceHttpMonitorsForCluster(clusterID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve existing HTTP monitors from Dynatrace for cluster %q: %w", clusterID, err)
+	}
+
+	if len(monitors) > 0 {
+		// Cleanup excess HTTP Monitors, if any exist
+		if len(monitors) > 1 {
+			err = removeDyntraceMonitors(dynatraceApiClient, monitors[1:])
+			if err != nil {
+				// log any errors regarding extra-monitor cleanup, but do not block further action
+				log.Error(err, "failed to cleanup excess Dynatrace monitors")
+			}
+		}
+
+		existingMonitor, err := dynatraceApiClient.GetDynatraceHttpMonitor(monitors[0].EntityId)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve existing monitor %q (ID=%q) from Dynatrace: %w", existingMonitor.Name, existingMonitor.EntityId, err)
+		}
+
+		if slices.Contains(existingMonitor.Locations, locationId) {
+			// Existing monitor matches expected - no further action needed
+			return nil
+		}
+		log.Info(fmt.Sprintf("monitor location needs to be updated, possibly due to API publishing strategy change in OCM. Deleting Dynatrace HTTP monitor %q in order to recreate in the correct synthetic location", existingMonitor.Name))
+		log.V(2).Info("current location(s) is %v, should be %q", existingMonitor.Locations, locationId)
+
+		err = dynatraceApiClient.DeleteSingleMonitor(existingMonitor.EntityId)
+		if err != nil {
+			return fmt.Errorf("failed to delete HTTP monitor %q (ID=%q) from Dynatrace: %w", existingMonitor.Name, existingMonitor.EntityId, err)
+		}
+	}
+
+	monitorId, err := dynatraceApiClient.CreateDynatraceHttpMonitor(monitorName, apiUrl, clusterID, locationId, clusterRegion)
+	if err != nil {
+		return fmt.Errorf("error creating HTTP monitor %q: %w", monitorName, err)
+	}
+	log.Info("Successfully created HTTP monitor", "monitor ID", monitorId)
+
+	return nil
+}
+
+func (r *HostedControlPlaneReconciler) deleteDynatraceHttpMonitorResources(dynatraceApiClient *dynatrace.DynatraceApiClient, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) error {
+	clusterId := hostedcontrolplane.Spec.ClusterID
+
+	err := dynatraceApiClient.DeleteDynatraceMonitorByCluserId(clusterId)
+	if err != nil {
+		return fmt.Errorf("error deleting HTTP monitor(s). Status Code: %v", err)
+	}
+	log.Info("Successfully deleted HTTP monitor(s)")
+	return nil
+}
+
+// RHOBSConfig holds RHOBS API configuration
+type RHOBSConfig struct {
```

Comment:
I realize this isn't your original code too, but should we split DT and RHOBS into their own files that synthetics.go just calls?  I can go either way, just thinking about organization/size